src/main/java/org/springframework/social/openidconnect/HttpClientFactory.java

```diff
@@ -114,4 +116,26 @@ private static X509HostnameVerifier getVerifier(boolean isStrict){
         return SSLSocketFactory.STRICT_HOSTNAME_VERIFIER;
     }
 
+    /**
+     * Gets the TLS version from jdk parameter.  Defaults to version 1.2 for JDK 1.7 if not provided
+     * @return
+     */
+    public static String getTlsVersion() {
+        // don't use "jdk.tls.client.protocols" as it was causing unexpected exception during unit testing
+        String tlsVer = System.getProperty("tlsProtocol");
+
+        if(StringUtils.isEmpty(tlsVer)) {
+            double javaVersion = Double.parseDouble(System.getProperty("java.specification.version"));
+            if(javaVersion == 1.7) {
```

Comment:
I don't think we should code per JDK.  There is no guarantee that within a major version TLS version will not be upgraded.  We should stick to the original plan of defaulting to TLSv1.2 and providing an option to override.
Is there any concerns of doing it that way?